### PR TITLE
Ignore default components check when a devfile references a parent

### PR DESCRIPTION
**Pull request by olexii4**
### What does this PR do?
Ignore default components check when a devfile references a parent.

### Screenshot/screencast of this PR

It needs for https://github.com/eclipse/che/issues/22103

### How to test this PR?

1. Create a new workspace from https://gist.githubusercontent.com/olexii4/fc3b757806e660dee0fbe1bc131df3d1/raw/1212b86a69440a30d3d06cbe412c72bbbb665daf/empty.yaml

```
schemaVersion: 2.2.0
metadata:
  generateName: empty
parent:
  id: udi
  registryUrl: https://registry.stage.devfile.io/
  version: 1.0.0
```

Знімок екрана 2023-04-18 о 16 15 30
### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Review by svor — APPROVED** [review]
LGTM

File: tools/devworkspace-generator/src/devfile/dev-container-component-finder.ts [hunk 1]
```diff
@@ -28,6 +28,9 @@ export class DevContainerComponentFinder {
 
     // only one, fine, else error
```
Comment (ibuziuk) [inline tools/devworkspace-generator/src/devfile/dev-container-component-finder.ts:29]: we probably should update or remove the comment above ^

Comment (olexii4) [reply tools/devworkspace-generator/src/devfile/dev-container-component-finder.ts]: I have fixed it

**Review by ibuziuk — APPROVED** [review]
@olexii4 :+1: 
could you please make sure that we backport the fix to 7.64.x for Dev Spaces 3.6

**Pull request merged**